pkgs/tools/misc/starfetch/default.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenv, lib, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "starfetch";
+  version = "0.0.1";
+
+  src = fetchFromGitHub {
+    owner = "Haruno19";
+    repo = "starfetch";
+    rev = version;
+    sha256 = "sha256-SwPEpulj1zPjr24M/op4IWRt+hvLjCgPNgamTLD7/GA=";
+  };
+
+  postPatch = ''
+    substituteInPlace ./src/starfetch.cpp --replace /usr/local/starfetch $out
```

Comment:
You should use makeFlags instead of patching

https://github.com/Haruno19/starfetch/blob/main/makefile#L4

Reply:
What does makeFlags have to do with the paths inside the source code? If starship.cpp is hardcoded to /usr/local how is makeFlags gonna change that?

Reply:
Ah its the cpp files.


If its the res dir it shouldn't be at the root of the $out

Reply:
I asked the developer about that too: https://github.com/Haruno19/starfetch/issues/2

Reply:
Should we tell him to put it somewhere in `/usr/share`?

Reply:
:shrug: not sure. I haven't even looked deep enough to work out if these resources are static or should be updated etc so I couldn't give an accurate answer

Reply:
> Should we tell him to put it somewhere in `/usr/share`?

So, would it be better to use `/usr/share/starfetch` rather than `/usr/local/starfetch` to store resource files (txt and json files)? 
There's no problem in changing that, but may I ask why?

Reply:
I'd imagine the argument is:

> /usr/share : Architecture-independent data

https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s11.html

Within `/usr/local` you could do `/usr/local/share/starfetch` but probably not `/usr/local/starfetch`

> No other directories, except those listed below, may be in /usr/local after first installing a FHS-compliant system.

see the directories section here

https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s09.html

So TL;DR `/usr/share/starfetch` or `/usr/local/share/starfetch` would be your options
And then I personally wouldn't add a `res` subdir since your res dir already is split nicely into 2 files and a dir

```
constellations/
help_message.txt
template
```

Reply:
Thank you for the exhaustive reply. 
I'll implement this changes as soon as possible, and then add a new release.

Reply:
[Release 0.0.2](https://github.com/Haruno19/starfetch/releases/tag/0.0.2) is now out. Files are now stored into `/usr/local/share/starfetch/` and the `res/` subdir has been removed.